File: surf-api-bukkit/surf-api-bukkit-api/src/main/java/dev/slne/surf/surfapi/bukkit/api/nms/bridges/packets/SurfBukkitNmsPacketBridges.java
package dev.slne.surf.surfapi.bukkit.api.nms.bridges.packets;

import dev.slne.surf.surfapi.bukkit.api.nms.SurfBukkitNmsBridge;
import dev.slne.surf.surfapi.bukkit.api.nms.bridges.packets.block.SurfBukkitNmsBlockPackets;
import dev.slne.surf.surfapi.bukkit.api.nms.bridges.packets.entity.SurfBukkitNmsSpawnPackets;
import dev.slne.surf.surfapi.bukkit.api.nms.bridges.packets.player.SurfBukkitNmsPlayerPackets;
import org.jetbrains.annotations.ApiStatus.NonExtendable;

@NonExtendable
public interface SurfBukkitNmsPacketBridges {

  SurfBukkitNmsSpawnPackets getSpawnPackets();

  PacketOperation createEmptyPacketOperation();

  SurfBukkitNmsBlockPackets getBlockPackets();

  SurfBukkitNmsPlayerPackets getPlayerPackets();

  static SurfBukkitNmsPacketBridges get() {
    return SurfBukkitNmsBridge.get().getPacketBridges();
  }
}

File: surf-api-bukkit/surf-api-bukkit-api/src/main/kotlin/dev/slne/surf/surfapi/bukkit/api/nms/bridges/packets/SurfBukkitNmsPacketBridges.kt
package dev.slne.surf.surfapi.bukkit.api.nms.bridges.packets

import dev.slne.surf.surfapi.bukkit.api.nms.NmsUseWithCaution
import dev.slne.surf.surfapi.core.api.util.requiredService

@NmsUseWithCaution
interface SurfBukkitNmsPacketBridges {
    fun createEmptyPacketOperation(): PacketOperation

    companion object {
        val instance = requiredService<SurfBukkitNmsPacketBridges>()
    }
}

@NmsUseWithCaution
val nmsPacketBridges get() = SurfBukkitNmsPacketBridges.instance